geowebcache/web/src/test/java/org/geowebcache/jetty/RestIntegrationTest.java

```diff
@@ -1042,6 +1047,104 @@ public void testLayerKillAll() throws Exception {
                 admin.getClient(), killCommand);
         TestCase.assertEquals(200, response.getStatusLine().getStatusCode());
     }
+    
+    @Test
+    public void testNewFileBlobstoreDontDeleteExistingContent() throws Exception {
+        // The directory already has stuff in it
+        File bsDir = temp.newFolder();
+        File existingFile1 = new File(bsDir, "existingFile1");
+        existingFile1.createNewFile();
+        File existingDirectory = new File(bsDir, "existingDirectory");
+        existingDirectory.mkdir();
+        File existingFile2 = new File(existingDirectory, "existingFile2");
+        existingFile2.createNewFile();
+        
+        String blobStore =
+                "<FileBlobStore>\n" +
+                "    <id>maliciousCache</id>\n" +
+                "    <enabled>true</enabled>\n" +
+                "    <baseDirectory>"+bsDir+"</baseDirectory>\n" +
+                "    <fileSystemBlockSize>4096</fileSystemBlockSize>\n" +
+                "</FileBlobStore>";
+        String layer1name = existingFile1.getName();
+        String layer1 =
+                "<wmsLayer>\n" +
+                "    <name>"+layer1name+"</name>\n" +
+                "    <blobStoreId>maliciousCache</blobStoreId>\n" +
+                "    <wmsUrl><string>http://example.com/geoserver/wms</string></wmsUrl>\n" +
+                "</wmsLayer>";
+        String layer1truncate =
+                "<truncateLayer><layerName>"+layer1name+"</layerName></truncateLayer>";
+        String layer2name = existingDirectory.getName();
+        String layer2 =
+                "<wmsLayer>\n" +
+                "    <name>"+layer2name+"</name>\n" +
+                "    <blobStoreId>maliciousCache</blobStoreId>\n" +
+                "    <wmsUrl><string>http://example.com/geoserver/wms</string></wmsUrl>\n" +
+                "</wmsLayer>";
+        String layer2truncate =
+                "<truncateLayer><layerName>"+layer2name+"</layerName></truncateLayer>";
+        
+        // Make it sure the blobstore doesn't exist already as that would invalidate the test
+        try (CloseableHttpResponse response = handleGet(URI.create("/geowebcache/rest/blobstores/maliciousCache.xml"), admin.getClient())){
+            assertThat(response, hasProperty("statusLine", hasProperty("statusCode", equalTo(404))));
+        }
+        
+        // Create a store
+        try (CloseableHttpResponse response = handlePut(URI.create("/geowebcache/rest/blobstores/maliciousCache"), admin.getClient(), blobStore)){
+            //assertThat(response, hasProperty("statusLine", hasProperty("statusCode", equalTo(201))));
```

Comment:
What's up with all these commented out asserts? 
I assume the exception is expected to be thrown from the handlePut, so shouldn't the body of the try have an Assert.fail()? Otherwise, uncomment or replace with a more descriptive comment.

Reply:
The response that comes back from these operations isn't important to the test, just that at the end the files that were there to start are still there. The commented out bits were to see what was going on while I set up the test and I should have deleted them.